org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/ObjectComparision.java

```diff
@@ -0,0 +1,654 @@
+			String v1 = objectExtraction(javaVal1);
+			String v2 = objectExtraction(javaVal2);
+			if (v1 == null || v2 == null) {
+				res = javaVal1.toString().equals(javaVal2.toString());
+				result.put(IMMEDIATE_RESULT_KEY, res);
+				return result;
+			}
+			res = v1.equals(v2);
+			if (!res) {
+				result.put(IMMEDIATE_RESULT_KEY, res);
+				String failureReason = "Selected objects are different ";
+				result.put(RESULT_FAILED, failureReason);
+				return result;
+			}
+			result.put(IMMEDIATE_RESULT_KEY, res);
+			return result;
+		}
+		result.put(IMMEDIATE_RESULT_KEY, res);
+		String failed = "Selected type is not IJavaValue type";
+		result.put(RESULT_FAILED, failed);
+		return result;
+	}
+	/**
+	 * Compare Array objects
+	 *
+	 * @param selectedObject1
+	 *            First selected object from variable view
+	 * @param selectedObject2
+	 *            Second selected object from variable view
+	 * @throws DebugException
+	 * @return returns a Map of comparison result details
+	 */
+	@SuppressWarnings("nls")
+	public Map<String, Object> compareArrays(IJavaVariable selectedObject1, IJavaVariable selectedObject2) throws DebugException {
+		Map<String, Object> result = new HashMap<>();
+		result.put(SELECTED_OBJECT_1, selectedObject1.getName());
+		result.put(SELECTED_OBJECT_2, selectedObject2.getName());
+		List<String> arrayElements1 = new ArrayList<>();
+		List<String> arrayElements2 = new ArrayList<>();
+		boolean res = false;
+		String val;
+		if (selectedObject1.getValue() instanceof IJavaValue javaVal1 && selectedObject2.getValue() instanceof IJavaValue javaVal2) {
+			for(IVariable jv : javaVal1.getVariables()) {
+				val = objectExtraction((IJavaValue) jv.getValue());
+				arrayElements1.add(val);
+			}
+			for (IVariable jv : javaVal2.getVariables()) {
+				val = objectExtraction((IJavaValue) jv.getValue());
+				arrayElements2.add(val);
+			}
+			result.put(VALUESET_1, arrayElements1);
+			result.put(VALUESET_2, arrayElements2);
+			return result;
+		}
+		result.put(IMMEDIATE_RESULT_KEY, res);
+		String failed = "Selected type is not IJavaValue type";
+		result.put(RESULT_FAILED, failed);
+		return result;
+	}
+
+	/**
+	 * Compare two String objects
+	 *
+	 * @param v1
+	 *            First selected object from variable view
+	 * @param v2
+	 *            Second selected object from variable view
+	 * @throws DebugException
+	 * @return returns a Map of comparison result details
+	 */
+	@SuppressWarnings("nls")
+	public Map<String, Object> stringCompare(IJavaVariable v1, IJavaVariable v2) throws DebugException {
+		Map<String, Object> result = new HashMap<>();
+		result.put(SELECTED_OBJECT_1, v1.getName());
+		result.put(SELECTED_OBJECT_2, v2.getName());
+		if (v1.getValue().getValueString().equals(v2.getValue().getValueString())) {
+			result.put(IMMEDIATE_RESULT_KEY, true);
+			return result;
+		}
+		result.put(IMMEDIATE_RESULT_KEY, false);
+		result.put(RESULT_FAILED, "Strings contains different values");
```

Comment:
Hard to understand. A String contains only characters. Maybe you mean "Strings are different"? If such message is displayed to users it should by NLS

Reply:
Yes, changed it